Framework/DataHandling/src/LoadSampleShape.cpp

```diff
@@ -189,20 +65,29 @@ void LoadSampleShape::exec() {
   }
 
   const std::string filename = getProperty("Filename");
-  std::ifstream file(filename.c_str());
-  if (!file) {
-    g_log.error("Unable to open file: " + filename);
-    throw Exception::FileError("Unable to open file: ", filename);
-  }
 
   const std::string filetype = filename.substr(filename.size() - 3);
 
   boost::shared_ptr<MeshObject> shape = nullptr;
+  ScaleUnits scaleType;
+  std::string scaleProperty = getPropertyValue("Scale");
+  if (scaleProperty == "m") {
```

Comment:
You have this code in multiple places. Could it be moved to a free function declared in the same place as the `ScaleUnits` enum?